File: coruscant/src/test/java/org/example/product/coruscant/test/CoruscantModuleTest.java
package org.example.product.coruscant.test;

import org.example.product.coruscant.CoruscantModule;
import org.junit.jupiter.api.Test;

import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStream;
import java.io.InputStreamReader;

import static org.assertj.core.api.Assertions.assertThat;

public class CoruscantModuleTest {

    @Test
    void testCoruscantModule() throws IOException {
        assertThat(new CoruscantModule()).isNotEqualTo(null);

        try(InputStream hello = CoruscantModuleTest.class.getResourceAsStream("testHello.txt")) {
            String message = new BufferedReader(new InputStreamReader(hello)).readLine();
            assertThat(message).isEqualTo("TEST HELLO!");
        }
    }
}

File: coruscant/src/test/kotlin/org/example/product/coruscant/test/CoruscantModuleTest.kt
package org.example.product.coruscant.test

import org.assertj.core.api.Assertions
import org.example.product.coruscant.CoruscantModule
import org.junit.jupiter.api.Test
import java.io.BufferedReader
import java.io.InputStreamReader

class CoruscantModuleTest {

    @Test
    fun testCoruscantModule() {
        Assertions.assertThat(CoruscantModule()).isNotEqualTo(null)
        CoruscantModuleTest::class.java.getResourceAsStream("testHello.txt").use { hello ->
            val message = BufferedReader(InputStreamReader(hello)).readLine()
            Assertions.assertThat(message).isEqualTo("TEST HELLO!")
        }
    }
}